sass/components/index-list.scss

```diff
@@ -0,0 +1,32 @@
+@import "mixins/index-list/index-list.scss";
+@import "mixins/index-list/index-list__item.scss";
+@import "mixins/index-list/index-list__item__index.scss";
+
+.index-list {
+  @include index-list;
+}
+
+.index-list__item {
+  @include index-list__item;
+}
+
+.index-list__item__index {
```

Comment:
Is this BEM?

Perhaps `.index-list-item__index` (list item as the block) or `.index-list__item-index` (index-list is still the block, with element item-index).

Mixin files should be renamed to match also.